src/common/mac/arch_utilities.cc

```diff
@@ -95,40 +95,33 @@ ArchInfo GetLocalArchInfo(void) {
 #ifdef __APPLE__
 
 std::optional<ArchInfo> GetArchInfoFromName(const char* arch_name) {
-  if (__builtin_available(macOS 13.0, iOS 16.0, *)) {
```

Comment:
If you have the latest SDK then you can make sure that `__builtin_available()` guards _at runtime_ for all target versions. There will be no issues at compile-time because you have everything right there in the SDK, but at runtime you can make sure that code only executes that is supported on the target-platform the code runs on.

But if you switch between two different APIs where one is deprecated in the latest SDK (`NXArchInfo`), but the other one isn't available in older SDKs (`mach-o/utils`), and you still want to support a build against the older SDK, then you need to check availability at compile time.

You can combine runtime-/compile-checks sometimes, but in this case I don't really see how it would make sense, because the targets and SDK availability fully overlap, especially since they also added `-Wdeprecated-declarations` to the ignored diagnostics. I will try to upstream this, maybe there is something I didn't consider yet.

Reply:
I wanted to explain the scenario where breakpad was built with a recent SDK, but the library executes on a device with an older system library (i.e., the scenario where an additional runtime check makes sense). So instead I decided to write the code. Thx for the push :-)